### PR TITLE
EventDeadLetters refactoring

### PR DESCRIPTION
**Pull request by hoangdat**
More details in https://ci.linagora.com/linagora/lgs/openpaas/james/issues/1072

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersContract.java [hunk 1]
```diff
@@ -115,68 +118,77 @@ static Event event(Event.EventId eventId) {
...
-
-            assertThat(eventDeadLetters.failedEventIds(GROUP_A).toStream())
-                .containsExactly(EVENT_ID_1);
+            eventDeadLetters.store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+            assertThat(eventDeadLetters.failedEvent(GROUP_A, INSERTION_ID_1).block())
+                .isEqualTo(EVENT_1);
         }
 
+//        @Test
+//        default void storeShouldIgnoreStoreDuplicatedEventsPerGroup() {
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersContract.java]: We need to handle this commented test.

Maybe we can be answering the question `what happens if I store 2 time the same event with a different insertionId?`

Then we might expect the event to be duplicated.

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/EventDeadLettersRoutes.java [hunk 2]
```diff
@@ -60,7 +61,7 @@
 public class EventDeadLettersRoutes implements Routes {
     private static final String BASE_PATH = "/events/deadLetter";
     private static final String GROUP_PARAM = ":group";
-    private static final String EVENT_ID_PARAM = ":eventId";
+    private static final String INSERTION_ID_PARAM = ":insertionId";
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/EventDeadLettersRoutes.java]: s/PARAM/PARAMETER/ please :-)

File: src/site/markdown/server/manage-webadmin.md [hunk 3]
```diff
@@ -2481,7 +2481,7 @@ Will return the full JSON associated with this event.
 Response codes:
 
  - 200: Success. A JSON representing this event is returned.
- - 400: Invalid group name or eventId
```
Comment (chibenwa) [inline src/site/markdown/server/manage-webadmin.md]: `Will return a list of event ids:` above is wrong. You should state: `Will return a list of insertionIds`

**Comment by hoangdat** [comment]
```
[c9c335fc46a431d1567a2b8d4b660cea07e51ac6] [ERROR] Errors: 
[c9c335fc46a431d1567a2b8d4b660cea07e51ac6] [ERROR]   MessageWithAttachmentMapperTest.messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsEmptyWhenNoAttachment » NoHostAvailable
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**